### PR TITLE
get bare service/nosetests running

### PR DESCRIPTION
**Pull request by Kjabraham**

### CODE REVIEW

File: service/models.py [hunk 1]
```diff
@@ -84,19 +83,21 @@ def deserialize(self, data):
...
-            self.date_added = data["date_added"]
+
         except KeyError as error:
-            raise DataValidationError("Invalid product: missing " + error.args[0])
-        except TypeError as error:
             raise DataValidationError(
-                "Invalid product: body of request contained" "bad or no data"
+                "Invalid Product: missing " + error.args[0]
+            )
+        except TypeError:
```
Comment (eleanor-mcneil) [inline service/models.py]: @Kjabraham the removed line above (line 97) has 'except TypeError as error:' while the added line just has 'except TypeError:' - what's the difference and why was 'as error' included in original code?

File: service/models.py [hunk 2]
```diff
@@ -113,18 +114,18 @@ def init_db(cls, app):
...
     def all(cls):
         """ Returns all of the products in the database """
-        logger.info("Processing all products")
+        logger.info("Processing all Products")
         return cls.query.all()
 
     @classmethod
     def find(cls, by_id):
-        """ Finds a product by its ID """
+        """ Finds a product by it's ID """
```
Comment (eleanor-mcneil) [inline service/models.py]: I changed this earlier, and the one below - 'its' is proper, which I know makes me a Grammar nerd/Nazi but...yeah.

File: service/service.py [hunk 3]
```diff
@@ -1,31 +1,78 @@
...
+GET /products - Returns a list all of the products
+GET /products/{id} - Returns the product with a given id number
+POST /products - creates a new product record in the database
+PUT /products/{id} - updates a product record in the database
+DELETE /products/{id} - deletes a product record in the database
 """
 
-import os
-import sys
-import logging
```
Comment (eleanor-mcneil) [inline service/service.py]: @Kjabraham why are these not needed anymore? I'm going to ask you to explain a bit so I can understand things before I merge the pull request ha.

Comment (Kjabraham) [reply service/service.py]: they just weren't used anywhere in that code so we do not need to import them.

File: service/models.py (hunk 1 above)
Comment (Kjabraham) [reply service/models.py]: the 'as error' sets the message the system throws to the variable 'error' however we then do not use that 'error' variable so it is unnecessary to declare. 

Hint: these are all changes coming from pylint running and trying to get us to 9/10

Comment (eleanor-mcneil) [reply service/models.py]: Ok sounds good thanks. I'll ask you later what pylint is.

File: service/service.py (hunk 3 above)
Comment (eleanor-mcneil) [reply service/service.py]: @Kjabraham is that expected that they're not used anywhere? Is that just for now or will they possibly be used as we go further along, and so would just commenting them out make more sense than deleting them entirely? Could you give me a one-two sentence explanation of what those lines would do?

Comment (Kjabraham) [reply service/service.py]: so those three lines are importing the libraries os, sys and logging. We use them in other files (models.py and __init__.py) for example so we import them there. I do not think we will ever need them in the service.py. The os and sys libraries are used to access things on the computer. We use 'os' to access the docker container which hosts our database for example.

Comment (eleanor-mcneil) [reply service/service.py]: Alright I will stem the continuing flow of questions and trust you that they're not needed here, and if so we can still add them back in afterwards.

**Pull request merged**